src/components/home/modals/ColdSignature.vue

```diff
@@ -168,8 +216,13 @@ export default {
                     delete data.api
                     delete data.opc
                     delete data.protocol
-                    data.timestamp *= 1e6
-                    if (transaction.isValidSignature(data, signature, this.dataObject.senderPublicKey, this.dataObject.transactionType) && !this.qrError) {
+                    if (data.hasOwnProperty('contractId') || data.hasOwnProperty('contract')) {
+                        data.fee = BigNumber(data.fee)
+                        data.timestamp = BigNumber(data.timestamp *= 1e6)
+                    } else {
+                        data.timestamp *= 1e6
+                    }
+                    if (((!data.hasOwnProperty('contractId') && transaction.isValidSignature(data, signature, this.dataObject.senderPublicKey, this.dataObject.transactionType)) || (data.hasOwnProperty('contractId') && transaction.isValidContractExecSignature(data, signature, data.senderPublicKey)) || (data.hasOwnProperty('contract') && transaction.isValidContractSignature(data, signature, data.senderPublicKey))) && !this.qrError) {
```

Comment:
is it better to use `transactionType` to separate the cases rather than use `hasOwnProperty('contractId')` and `data.hasOwnProperty('contract')`?